.github/workflows/helm-ci-test.yml

```diff
@@ -0,0 +1,101 @@
+          repo=${repo#"https://"} # trim prefix
+
+          if [ "$repo" != "github.com/Azure/azure-service-operator" ]; then
+            echo "Skipping 'go get' check for fork…" 
+            exit 0
+          fi
+
+          if [ -z "$sha" ]; then 
+            # this means we were triggered by push to 'main', 
+            # not a PR, so use the sha that triggered this
+            sha='${{ github.sha }}'
+          fi
+
+          go get "$repo/v2@$sha"
+
+  test-generator:
+    runs-on: ubuntu-latest
+    permissions:
+      packages: read
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0 # required to access tags
+          submodules: 'true'
+
+      - name: Log in to GitHub Docker Registry
+        uses: docker/login-action@v1
+        with:
+          registry: docker.pkg.github.com # ghcr.io not yet enabled for Azure org
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      # Note: Changes to this step must also be mirrored into pr-validation-with-secrets.yaml
+      - name: Build devcontainer image
+        # We must issue a manual pull before the build so the image gets copied locally, because
+        # docker.pkg.github.com is not a valid Docker registry and doesn't work with --cache-from,
+        # however, `docker pull` will fall back to other methods that do work and get the image loaded.
+        #
+        # This message comes from "docker pull":
+        #
+        # Run docker pull docker.pkg.github.com/azure/azure-service-operator/aso-devcontainer:latest
+        # WARNING: ⚠️ Failed to pull manifest by the resolved digest. This registry does not
+        #  appear to conform to the distribution registry specification; falling back to
+        #  pull by tag.  This fallback is DEPRECATED, and will be removed in a future
+        #  release.  Please contact admins of https://docker.pkg.github.com. ⚠️
+        #
+        # See: https://github.com/moby/moby/issues/41687#issuecomment-733826074 and related issues
+        run: |
+          docker pull docker.pkg.github.com/azure/azure-service-operator/aso-devcontainer:latest
+          docker build --cache-from docker.pkg.github.com/azure/azure-service-operator/aso-devcontainer:latest --tag devcontainer:latest .devcontainer
+        env:
+          DOCKER_BUILDKIT: 1
+
+      - name: Run devcontainer image
+        id: devcontainer
+        run: |
+          container_id=$(docker create -w /workspace -v $GITHUB_WORKSPACE:/workspace -v /var/run/docker.sock:/var/run/docker.sock --network=host devcontainer:latest)
+          docker start "$container_id"
+          echo "::set-output name=id::$container_id"
+
+      - name: Spin up a kind cluster
+        run: |
+          container_id=${{steps.devcontainer.outputs.id}}
+          docker exec "$container_id" task controller:kind-create
+
+      - name: Test Helm Installation
```

Comment:
This isn't going to work for forks, unless you set up the integration to make it work for forks.

I am wondering if it makes more sense to, instead of having a separate CI job for Helm, to just use the `test-integration-kind` job, which I've already set up to work for fork PRs. You could do this by creating a new `controller:kind-create-helm-podidentity`, and then substitute that into the `controller:test-integration-kind` and `controller:test-integration-kind-ci` targets

Reply:
Then we'd have 3 sets of validation we do (rather than 4 as you've proposed):

1. envtest with recordings. Needs no Azure secrets and runs pretty fast.
2. kind cluster with Helm chart installed on it (using the latest master container image and YAMLs)
3. kind cluster in multitenant mode (installed manually)

Otherwise if we do not combine things, we'd have:

item 2 == kind cluster with latest operator installed manually
item 4 == kind cluster with latest Helm chart

and I'm not sure there's a lot of practical difference between these... especially because we already test installation of the standalone operator in the multitenant deployment as well (just added by @babbageclunk)

Reply:
Yes, sounds very clean. I've made changes according to the suggestion.